File: matlab/fnc/extractFeature.m
%
% extractFeature        - generates a biometric template from an iris in an
% eye image.
%
% Arguments:
%	eyeimage_filename   - the file name of the eye image
%
% Output:
%	template		    - the binary iris biometric template
%	mask			    - the binary iris noise mask
%

function [template, mask] = extractFeature(eyeimage_filename)
%% Setup
% Normalisation parameters
radial_res = 20;
angular_res = 240;

% Feature encoding parameters
nscales=1;
minWaveLength=18;
mult=1;
sigmaOnf=0.5;


%% Perform segmentation
eyeimage = imread(eyeimage_filename);
[circleiris, circlepupil, imagewithnoise] = segmentiris(eyeimage);


%% Perform normalization
[polar_array, noise_array] = normaliseiris(imagewithnoise, ...
    circleiris(2), circleiris(1), circleiris(3), ...
    circlepupil(2), circlepupil(1), circlepupil(3), ...
    eyeimage_filename, radial_res, angular_res);


%% Perform feature encoding
[template, mask] = encode(polar_array, noise_array, nscales, minWaveLength, mult, sigmaOnf); 


end

File: python/fnc/extractFeature.py
##-----------------------------------------------------------------------------
##  Author      : Chinh-Thuy Nguyen
##  Date        : Nov 11, 2017
##
##  Interpreter : 3.5
##  Editor      : Sublime Text 3
##-----------------------------------------------------------------------------

##-----------------------------------------------------------------------------
##  Import
##-----------------------------------------------------------------------------
from cv2 import imread

from fnc.segment import segment
from fnc.normalize import normalize
from fnc.encode import encode


##-----------------------------------------------------------------------------
##  Function
##-----------------------------------------------------------------------------
def extractFeature(im_filename):
	"""
	Description:
		Extract features from an iris image

	Input:
		im_filename	- The input iris image

	Output:
		template	- The extracted template
		mask		- The extracted mask
		im_filename	- The input iris image
	"""
	# Normalisation parameters
	radial_res = 20
	angular_res = 240

	# Feature encoding parameters
	minWaveLength = 18
	mult = 1
	sigmaOnf = 0.5

	# Perform segmentation
	im = imread(im_filename, 0)
	ciriris, cirpupil, imwithnoise = segment(im)

	# Perform normalization
	polar_array, noise_array = normalize(imwithnoise, ciriris[1], ciriris[0], ciriris[2],
										 cirpupil[1], cirpupil[0], cirpupil[2],
										 radial_res, angular_res)

	# Perform feature encoding
	template, mask = encode(polar_array, noise_array, minWaveLength, mult, sigmaOnf)

	# Return
	return template, mask, im_filename

